pkg/security/http/auth.go

```diff
@@ -58,9 +58,9 @@ type Authenticator interface {
 // returns a decision if the authorization passed
 //go:generate counterfeiter . Authorizer
 type Authorizer interface {
-	// Authorize returns decision specifying
-	// whether the authorizer ran or not and an error if one occurs
-	Authorize(req *web.Request) (Decision, error)
+	// Authorize returns decision specifying whether the authorizer allowed, denied or abstained from giving access,
+	// the access level associated with the decision and an error if one occurs
```

Comment:
Which access level should I return if I'm implementing an authorizer? The one in the user context or the one required to perform the operation? What am I expected to do with this access level if I'm a consumer of an authorizer?

Reply:
the consumer is the authz filter that sets it in the context currently. So in the context that your authorizer would accept the only possible value for the AccessLevel will be whatever is the zero value used (or whatever the authentication layer set when creating the inital context)